mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go

```diff
@@ -255,22 +255,24 @@ func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	for _, fi := range ti.FamilyInfos {
-		if fi.Name == columnFamily {
-			d.SetId(fi.GCPolicy)
-			// Only set `gc_rules`` when the legacy fields are not set. We are not planning to support legacy fields.
-			if d.Get("mode") == "" && d.Get("max_age") == "" && d.Get("max_version") == "" {
-				gcRuleString, err := gcPolicyToGCRuleString(fi.FullGCPolicy, true)
-				if err != nil {
-					return err
-				}
-				gcRuleJsonString, err := json.Marshal(gcRuleString)
-				if err != nil {
-					return fmt.Errorf("Error marshaling GC policy to json: %s", err)
-				}
-				d.Set("gc_rules", string(gcRuleJsonString))
+		if fi.Name != columnFamily {
+			continue
+		}
+
+		d.SetId(fi.GCPolicy)
+		// Only set `gc_rules`` when the legacy fields are not set. We are not planning to support legacy fields.
+		if d.Get("mode") == "" && d.Get("max_age") == "" && d.Get("max_version") == "" {
```

Comment:
Due to how empty values are stored in state, and the types for max_age & max_version I don't think this can ever be true.

Have you been able to test this functionality out locally and see it work correctly?

Reply:
Good catch Sam! I forgot about the fact that max_age and max_version are lists.

I only compared the local `gc_rules` and the remote `gc_rules`. Since they are same even though the code didn't set it, the tests still passed.

I talked with Riley earlier regarding testing this. He said this is something hard to test. I think mainly because GC policy doesn't have import support.

I verified by checking the logs. I am going to try something something else locally to verify. Perhaps directly in KCC. Let me know if you have a better idea. Thanks!

Reply:
Here is the manual testing I have done.

Without this PR:
1. tf apply
2. Modify gc_rules in both config and state files. For example: "10h" to "12h"
3. tf plan showed no diff

Above shows no diff because both config file and the state file match and there is no refresh (the bug we are trying to fix) from the remote resource.

Without PR:
1. tf apply
2. Modify gc_rules in both config and state files. For example: "10h" to "12h"
3. tf plan showed diff (will apply max_age = "10h" -> "12h")

We are seeing diff because we are reading the remote gc policy back now.